### PR TITLE
add: okd-rpms buildconfig and tools:fedora37 istag;  change: fedora-coreos buidconfig to point to git ref release-4.12, build-from-scrathc and full-rebuild pipeline to include okd-rpms build

### PR DESCRIPTION
**Pull request by dsevost**
hello dear okd-team, please review pull-req

tldr:
added ability to build `okd-rpms` as a part of the pipeline `build-from-scratch`

details:
- at the moment, the assisted service is still looking for the okd-rpms image tag in the `registry/okd/okd-release:4.12.0`, so its added `variants/fcos/40-okd-rpms.yaml`, with refers to `istag tools:fedora37`
- to be able to build `okd-rpms` in the main pipeline (sorry not able to find better way at the moment than refer of array indices like `3` and `0` via hardcoded style), it needs to be fixed `build-from-scratch` pipeline (task `batch-03`), also i'm not sure the pipeline `full-rebuild` is not out of date, since i've not found `machine-os-content` buildconfig, although found `machine-os-content' istag
- all buildconfigs (except 03-forked-dockerfiles.yaml, 04-builder.yaml, 10-branding.yaml, 30-fedora-coreos.yaml) reffer to git branch `release-4.12` and it has some side effetcs for the bc `fedora-coreos` (30-fedora-coreos.yaml) which belogs of tools:fedora-coreos, wich points to quay.io/coreos-assembler/fcos:next-devel; and i'm not sure that fedora-coreos:next-devel is the best choice for 4.12 (cri-o 1.28), it seems fedora-coreos:stable most suitable at the moment? lets fix this by changing `ref` attribute of 30-fedora-coreos.yaml builconfig to git ref `release-4.12`

### CODE REVIEW

File: variants/fcos/okd-rpms/manifests/99-okd-master-disable-mitigations.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: machineconfiguration.openshift.io/v1
```
Comment (vrutkovs) [inline variants/fcos/okd-rpms/manifests/99-okd-master-disable-mitigations.yaml:1]: I don't think these manifests are needed in `okd-rpms`, which should just contain kubelet+crio RPMs.
We do need them for `okd-machine-os`

File: variants/fcos/okd-rpms/manifests/99-okd-worker-disable-mitigations.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+apiVersion: machineconfiguration.openshift.io/v1
```
Comment (vrutkovs) [inline variants/fcos/okd-rpms/manifests/99-okd-worker-disable-mitigations.yaml:1]: ditto

File: variants/fcos/okd-rpms/manifests/99-sca-disable.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+kind: Secret
```
Comment (vrutkovs) [inline variants/fcos/okd-rpms/manifests/99-sca-disable.yaml:1]: ditto

Comment (dsevost) [reply variants/fcos/okd-rpms/manifests/99-sca-disable.yaml:1]: yes, it was just copy-pasted from registry/okd/okd-release:4.12.0;
while mitigations looks reasonable, then what is `sca` - i have no idea.
should remove them all?

File: variants/fcos/okd-rpms/manifests/99-okd-master-disable-mitigations.yaml (hunk 1 above)
Comment (dsevost) [reply variants/fcos/okd-rpms/manifests/99-okd-master-disable-mitigations.yaml]: done

File: variants/fcos/okd-rpms/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+FROM release:artifacts as artifacts
+FROM release:machine-config-operator as mcd
```
Comment (aleskandro) [inline variants/fcos/okd-rpms/Dockerfile:2]: why do we need this layer?

File: variants/fcos/okd-rpms/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+FROM fedora:37
+ARG CRIO_VERSION=1.25
+
+WORKDIR /rpms
+
+COPY --from=artifacts /srv/repo/* /tmp/rpms
+
+RUN \
+    mkdir -p /rpms && \
+    mv -v /tmp/rpms/$([ -d /tmp/rpms/$(uname -m) ] && echo $(uname -m)/)*.rpm /rpms/ && \
```
Comment (aleskandro) [inline variants/fcos/okd-rpms/Dockerfile:13]: is it worth adding the rpms in the `artifacts` and adding an imagestreamtag referencing it with name okd-rpms?

In future, the two names should converge.

File: variants/fcos/okd-rpms/Dockerfile (hunk 4 above)
Comment (dsevost) [reply variants/fcos/okd-rpms/Dockerfile:2]: mcd layer might be unnecessary, it was introduced just for "backward compatibility" and used for
```
# TODO: check: is it really needed
COPY --from=mcd /usr/bin/machine-config-daemon /binaries/machine-config-daemon
```

File: variants/fcos/okd-rpms/Dockerfile (hunk 5 above)
Comment (dsevost) [reply variants/fcos/okd-rpms/Dockerfile:13]: wouldn't like to break any builds which depend on original `quay.io/coreos-assembler/fcos:next-devel`, if any

Comment (dsevost) [reply variants/fcos/okd-rpms/Dockerfile:13]: as i understand correctly scos would be like rhel coreos and it will contain any necessary rpms (including crio), its a reason when fcos variant is a little separated at the momemnt (or opposite)

Comment (aleskandro) [reply variants/fcos/okd-rpms/Dockerfile:13]: The artifacts image is not part of the payload but an auxiliary image. 

Adding the cri-o rpms and using some symbolic links to refer the rpms where they are currently stored should be enough if I understand correctly the use case. The consumers of the rpms will use the rpms at need... 

https://github.com/okd-project/okd-payload-pipeline/blob/main/forks/artifacts/Dockerfile.centos9

Comment (dsevost) [reply variants/fcos/okd-rpms/Dockerfile:13]: there are still some references to `GetOKDRPMSImage` within assisted service codebase:
```
https://github.com/openshift/assisted-service/blob/99125bfa925143394e32f785e104c63624d1f601/internal/oc/release.go#L45
https://github.com/openshift/assisted-service/blob/99125bfa925143394e32f785e104c63624d1f601/internal/ignition/ignition.go#L1627
```
finally the `GetOKDRPMSImage` refers to `okdRPMSImageName  = "okd-rpms"` as tag for registry/okd/okd-contents:4.12.0-0.okd-xxxx-okd-rpms
and this is a point where scos might be more suitable: full control for OS version/build and ostree (including crio, kubelet, cli, ncat, etc) without temporary rpm overlaying on bootkube/bootstrap procedure

**Comment by dsevost** [comment]
included into https://github.com/okd-project/okd-payload-pipeline/pull/39#issue-1746710513
